### DISCUSSION TITLE
SQLAlchemy models are not able to pick up session's schema while getting values

### DISCUSSION THREAD

**Discussion by pdudhaiya-crest** [sqlalchemy/sqlalchemy#10324] [Usage Questions]
### Describe the bug

I am trying to use a single ORM model for all tenants in my application. The reason is, the table is exactly the same. It just needs to be in a separate schema so that the tables are secluded from the application's tables as well as other user's tables.

for example, the table **integrations** belongs to schema **tenant1** as well as **tenant2** Thus, I should be able to use the same model that I have created - Integration to be used for both tenants.

In my FastAPI endpoint, each request will open its own session. Thus what I do is -

```py
from models import Integration
db_session.connection(
    execution_options={"schema_translate_map": {None: "tenant1"}}
)
integration_object = Integration(*args)
db_session.add(integration_object)
db_session.commit()

# the following does not work
db_session.refresh()
print(integration_object.id)
```

The error is thrown from db_session.refresh(). The SQLAlchemy is able to write the item to the database. However, while retrieving it internally it does not take into account the schema that was set in line 2 and tries to search in public schema.

Any ideas?

### SQLAlchemy Version in Use

2.0.20

### DBAPI (i.e. the database driver)

psycopg2

### Database Vendor and Major Version

PostgreSQL 12

### Python Version

3.10

### Operating system

Ubuntu

### To Reproduce

```python
from models import Integration
db_session.connection(
    execution_options={"schema_translate_map": {None: "tenant1"}}
)
integration_object = Integration(*args)
db_session.add(integration_object)
db_session.commit()

# the following does not work
db_session.refresh()
print(integration_object.id)
```

### Error

```
E           app.database.exception.DBException: (psycopg2.errors.UndefinedTable) relation "integrations" does not exist
E           LINE 2: FROM integrations 
E                        ^
E
E           [SQL: SELECT integrations.id, integrations.name, integrations.integration_id, integrations.execution_path, integrations.status, integrations.created_at, integrations.updated_at, integrations.integration_schema, integrations.version, integrations.author, integrations.nist_framework, integrations.category 
E           FROM integrations 
E           WHERE integrations.id = %(pk_1)s]
E           [parameters: {'pk_1': 22}]
E           (Background on this error at: https://sqlalche.me/e/20/f405)app\database\crud\integration_crud.py:140: DBException
```

**Comment by CaselIT** [comment]
Hi,

That code is not runnable. How it db_session procured? Is the connection call the fist db action done by the session?

**Reply by CaselIT** [reply]
Also that's the documented behaviour: https://docs.sqlalchemy.org/en/20/orm/session_transaction.html#setting-isolation-for-individual-transactions
see note after the `commit()`
